pkg/driver/controller.go

```diff
@@ -709,6 +710,12 @@ func (d *ControllerService) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		case FastSnapshotRestoreAvailabilityZones:
 			f := strings.ReplaceAll(value, " ", "")
 			fsrAvailabilityZones = strings.Split(f, ",")
+		case OutpostArnKey:
```

Comment:
Nit: Would the validateCreateSnapshotRequest helper function be a better place for this arn check? (Due to having one place in the code where most of the InvalidArgument errors are created) 

The function definition is on line 790

Reply:
In my opinion errors from parameters should be in the switch statement, that is consistent with the code and reduces duplicate parsing.

Reply:
Where else in the switch statement are errors returned? I don't see any such code consistency. 

Seems to me that switch statement is for parsing, not for validation. 

And wouldn't we want to not make a needless call to describeSnapshots (I understand that we batch, but still).

Reply:
> In my opinion errors from parameters should be in the switch statement, that is consistent with the code and

The only error from parameters that is in the switch statement is the error one receives for using a non-accepted parameter. I think it's cleaner to have the function named `validateCreateSnapshotRequest` validate as much of the request as possible before any AWS API calls are wastefully made (e.g. DescribeSnapshots), than to have yet another place an error can be returned. 

But given that this is a nit, I'm fine with this conversation being resolved.